Cython/Compiler/PyrexTypes.py

```diff
@@ -3107,6 +3131,10 @@ def __init__(self, return_type, args, has_varargs = 0,
         self.args = args
         self.has_varargs = has_varargs
         self.optional_arg_count = optional_arg_count
+        if isinstance(exception_value, (int, float, str, bytes)):
+            # happens within Cython itself when writing custom function types
+            # for utility code functions.
+            exception_value = CFuncType.ExceptionValue(exception_value, str(exception_value))
         self.exception_value = exception_value
```

Comment:
Just to be sure we're not missing a type above.
Alternative 1:
```suggestion
        else:
            assert isinstance(exception_value, self.ExceptionValue)
        self.exception_value = exception_value
```